docs: new channel concept doc (#1007)
asyncapi/website: pages/docs/concepts/channel.md

---
title: Channel
weight: 20
---

# What is a channel?
A channel is a mechanism that is created by the server for the organization and transmission of messages. 

# What is the purpose of a channel?
Channels play a crucial role when it comes to the communication between producers and consumers. A producer can send a message through the channel and the consumer receives messages from a particular channel.

```mermaid
graph LR
    A[Producer] --> B(message)
    B --> C(Channel)
    C --> D[Consumer]
    C --> E[Consumer]
    C --> F[Consumer]

> ```

Author: Hi @derberg @smoya @fmvilas this is what i could come up in terms of the `channel` spec term. Any suggestions on improving this and also the diagram as well.

Reviewer: Looks good to me. I'm missing some more information, especially for EDA newcomers. A channel is usually known as "topic", "queue", "routing key", "path", or "subject", and I'm sure there are other terms 😅 

You did a great job explaining it's for the categorization of messages. It's also used for routing purposes, so the right messages end up in the right consumers.

Reviewer 2: yup, agree 💯 that it would be good to add maybe a new section, on how `channel` relates to AsyncAPI document. That it is the same as topic and others, depending what protocol someone is using

Author: Okay i have made some changes @derberg , quick question is the diagram okay though?

---
title: Channel
weight: 20
---

# What is a channel?

> A `channel` is a mechanism that is created by the server for the organization and transmission of messages. Channels can also be defined as a _topic ,queue, routing key, path, or subject_ depending on the protocol being used.

Reviewer 3: ```suggestion
A `channel` is a mechanism created by the server for the organization and transmission of messages. Users can define channels as a _topic, queue, routing key, path,_  or _ subject_ depending on the protocol used.
```

> # What is the purpose of a channel?

Reviewer 3: ```suggestion
# Why do we need channels?
```

Sorry for the confusion Thulie, we're going with a `why` question in this heading instead 😅

> Channels play a crucial role when it comes to the communication between `producers` and `consumers`. A producer can send a message through the channel and the consumer receives messages from a particular channel. A channel's sole purpose is to make sure that the right messages route to the right consumers.

Reviewer 3: ```suggestion
Channels play a crucial role in communication between `producers` and `consumers`. A producer can send a message through the channel, and the consumer receives messages from a particular channel. A channel's sole purpose is to ensure the right messages route to the right consumers.
```

```mermaid
graph LR
    A[Producer] --> B(message)
    B --> C(Channel)
    C --> D[Consumer]
    C --> E[Consumer]
    C --> F[Consumer]
```

> The diagram above shows the communication between the producer and the consumer. The producer sends a message through the channel and the channel queues the message to the specific consumer.

Reviewer 3: ```suggestion
The diagram above shows the communication between the producer and the consumer. The producer sends a message through the channel, and the channel queues the message to the specific consumer.
```
